module/platform/Webots/src/Webots.cpp

```diff
@@ -252,7 +252,8 @@ namespace module::platform {
 
         on<Configuration>("WebotsCameras").then([this](const Configuration& config) {
             // Strip the .yaml off the name of the file to get the name of the camera
-            const std::string name = ::basename(config.fileName.substr(0, config.fileName.find_last_of('.')).c_str());
+            const std::string config_file = config.fileName.string();
+            const std::string name        = ::basename(config_file.substr(0, config_file.find_last_of('.')).c_str());
```

Comment:
I think this logic should go in a method of Configuration, something like `[[nodiscard]] string get_name() const;`

Reply:
Great suggestion. Doing that now. Thanks